services/ui-src/src/components/fields/TextField.test.tsx

```diff
@@ -68,6 +72,40 @@ describe("<TextField />", () => {
     });
   });
 
+  describe("Text field hide condition logic", () => {
+    test("Text field is hidden if its hide conditions' controlling element has a matching answer", async () => {
+      mockGetValues({
+        elements: [
+          {
+            answer: "yes",
+            type: "reportingRadio",
+            label: "Should we hide the other radios on this page?",
+            id: "reporting-radio",
+          },
+        ],
+      });
+      render(textFieldComponent);
+      const textField = screen.queryByLabelText("test label");
+      expect(textField).toBe(null);
+    });
+
+    test("Text field is NOT hidden if its hide conditions' controlling element has a different answer", async () => {
+      mockGetValues({
+        elements: [
+          {
+            answer: "idk",
+            type: "reportingRadio",
+            label: "Should we hide the other radios on this page?",
+            id: "reporting-radio",
+          },
+        ],
+      });
+      render(textFieldComponent);
+      const textField = screen.queryByLabelText("test label");
```

Comment:
My suggestion above applies here too:
```suggestion
      const textField = screen.getByLabelText("test label");
```